### PR TITLE
Reduce warnings in test suite for new return type system

### PR DESCRIPTION
**Pull request by dwierichs**
**Context:**
There are quite a few warnings raised when running the tests in `tests/returntypes`.

**Description of the Change:**
This PR removes a part of the warnings, in particular those that are based on the test suite itself.

**Benefits:**
Fewer warnings

**Possible Drawbacks:**
N/A

**Related GitHub Issues:**
#3428

### CODE REVIEW

File: tests/returntypes/gradients/test_jvp.py [hunk 1]
```diff
@@ -436,20 +436,18 @@ def test_prob_expectation_values(self, tol):
         assert isinstance(res, tuple)
         assert len(res) == 2
 
-        expected = (
+        expected = [
```
Comment (rmoyard) [inline tests/returntypes/gradients/test_jvp.py:439]: Thanks great catch :100:

File: tests/returntypes/transforms/test_metric_tensor_new.py [hunk 2]
```diff
@@ -123,12 +124,11 @@ def test_construct_subcircuit(self):
         assert isinstance(tapes[1].operations[3], qml.Hadamard)
 
         # third parameter subcircuit
-        assert len(tapes[2].operations) == 4
+        assert len(tapes[2].operations) == 3
```
Comment (rmoyard) [inline tests/returntypes/transforms/test_metric_tensor_new.py:127]: What happened here?

File: tests/transforms/test_metric_tensor.py [hunk 3]
```diff
@@ -703,7 +703,7 @@ def test_no_trainable_params_tape(self):
         assert res == ()
 
 
-fixed_pars = np.array([-0.2, 0.2, 0.5, 0.3, 0.7], requires_grad=False)
+fixed_pars = [-0.2, 0.2, 0.5, 0.3, 0.7]
```
Comment (rmoyard) [inline tests/transforms/test_metric_tensor.py:706]: Is it raising a warning?

**Review by rmoyard — COMMENTED** [review]
Thanks @dwierichs, it looks good :100: I have a couple of questions, happy to approve after that!

File: tests/returntypes/transforms/test_metric_tensor_new.py (hunk 2 above)
Comment (dwierichs) [reply tests/returntypes/transforms/test_metric_tensor_new.py:127]: Because of the `return` statement in the tape construction, this test was not executed at all. It simply ended at the line 
```python
return qml.expval(qml.PauliX(0)), qml.expval(qml.PauliX(1))
```
because there is no internal function to end, but the function `test_construct_subcircuit` :grimacing:

File: tests/transforms/test_metric_tensor.py (hunk 3 above)
Comment (dwierichs) [reply tests/transforms/test_metric_tensor.py:706]: Yes, because the `fubini_ansatz#` ansatze are used with the interfaces, and using Autograd parameters mixed with TF/JAX/Torch parameters raises a warning in the `math/utils.py` file when using multi-dispatch functions.
